scripts/mod_loader/modapi/palette.lua

```diff
@@ -0,0 +1,433 @@
+
+local VANILLA_PALETTE_ID = {
+	GetText("Squad_Archive_A"),
+	GetText("Squad_Rust_A"),
+	GetText("Squad_Pinnacle_A"),
+	GetText("Squad_Detritus_A"),
+	GetText("Squad_Archive_B"),
+	GetText("Squad_Rust_B"),
+	GetText("Squad_Pinnacle_B"),
+	GetText("Squad_Detritus_B"),
+	GetText("Squad_Secret")
+}
+
+local VANILLA_PALETTE_NAME = {
+	GetText(VANILLA_PALETTE_ID[1].." Olive"),
+	GetText(VANILLA_PALETTE_ID[2].." Orange"),
+	GetText(VANILLA_PALETTE_ID[3].." Blue"),
+	GetText(VANILLA_PALETTE_ID[4].." Yellow"),
+	GetText(VANILLA_PALETTE_ID[5].." Shivan"),
+	GetText(VANILLA_PALETTE_ID[6].." Red"),
+	GetText(VANILLA_PALETTE_ID[7].." Blue"),
+	GetText(VANILLA_PALETTE_ID[8].." Tan"),
+	GetText(VANILLA_PALETTE_ID[9].." Purple")
+}
+
+local COLOR_NAME_2_INDEX = {
+	lights = 1,
+	main_highlight = 2,
+	main_light = 3,
+	main_mid = 4,
+	main_dark = 5,
+	metal_dark = 6,
+	metal_mid = 7,
+	metal_light = 8,
+	PlateHighlight = 1,
+	PlateLight = 2,
+	PlateMid = 3,
+	PlateDark = 4,
+	PlateOutline = 5,
+	PlateShadow = 6,
+	BodyColor = 7,
+	BodyHighlight = 8
+}
+
+local canAddPalettes = true
+
+local PaletteDictionary = {
+	_dictionary = {},
+	_array = {},
+	add = function(self, id, palette)
+		assert(type(id) == 'string')
+		assert(type(palette) == 'table')
+
+		local index = #self._array + 1
+
+		palette.id = id
+		palette.imageOffset = index
+		palette.imageOffsetOrig = index
+
+		self._dictionary[id] = palette
+		self._array[#self._array+1] = palette
+	end,
+
+	get = function(self, a)
+		if type(a) == 'number' then
+			return self._array[a]
+		else
+			return self._dictionary[a]
+		end
+	end,
+
+	swap = function(self, a, b)
+		local paletteA, paletteB
+		if type(a) == 'number' then
```

Comment:
Shouldn't you use the `get` function defined above here? No reason to duplicate the logic